ktor-client/ktor-client-core/common/src/io/ktor/client/utils/SharedCollections.kt

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright 2014-2020 JetBrains s.r.o and contributors. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package io.ktor.client.utils
+
+import io.ktor.util.*
+import io.ktor.util.collections.*
+
+internal fun <K : Any, V : Any> sharedMap(): MutableMap<K, V> {
+    if (PlatformUtils.IS_NATIVE) {
+        return ConcurrentMap()
+    }
+
+    return mutableMapOf()
```

Comment:
Is a mutable map thread-safe and can be shared safely?

Reply:
`Shared` doesn't mean `Concurrent` here.

Reply:
This configuration shouldn't be used concurrently but can be accessed from the different threads, so it should be shared on native and can be usual on the JVM

Reply:
For the native, it's called`ConcurrentMap`. Maybe we should think about a more appropriate name?

Reply:
The native map is `Concurrent`(synchronized) and  `ConcurrentMap` is shared. We have no separate `shared` map in native, so it's ok to use concurrent here.